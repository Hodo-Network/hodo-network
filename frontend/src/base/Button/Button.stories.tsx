import { Story, Meta } from "@storybook/react";

import { Button, ButtonProps } from "./Button";

export default {
  title: "UI/Button",
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => (
  <Button {...args}>Button</Button>
);

export const Primary = Template.bind({});
Primary.args = {
  color: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  color: "secondary",
};

export const Large = Template.bind({});
Large.args = {
  size: "lg",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "md",
};

export const Small = Template.bind({});
Small.args = {
  size: "sm",
};
