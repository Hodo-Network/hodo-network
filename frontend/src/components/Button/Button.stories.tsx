import React from "react";
import { Story, Meta } from "@storybook/react";

import { Button, ButtonProps } from "./Button";

export default {
  title: "UI/Button",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
  },
} as Meta;

const Template: Story<ButtonProps> = (args) => (
  <Button {...args}>Button</Button>
);

export const Primary = Template.bind({});
Primary.args = {
  primary: true,
};

export const Secondary = Template.bind({});
Secondary.args = {};

export const Large = Template.bind({});
Large.args = {
  size: "large",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "medium",
};

export const Small = Template.bind({});
Small.args = {
  size: "small",
};
