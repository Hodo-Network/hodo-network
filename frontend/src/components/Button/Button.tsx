import React from "react";

export interface ButtonProps {
  primary?: boolean;
  size?: "small" | "medium" | "large";
  onClick?: () => void;
  disabled?: boolean;
  backgroundColor?: string;
}

export const Button: React.FC<ButtonProps> = ({
  primary = false,
  size = "medium",
  disabled = false,
  backgroundColor,
  ...props
}) => {
  const mode = primary ? "primary" : "secondary";
  return (
    <button
      type='button'
      className={["btn", `btn--${mode}`, `btn--${size}`].join(" ")}
      style={{ backgroundColor }}
      disabled={disabled}
      {...props}
    />
  );
};
