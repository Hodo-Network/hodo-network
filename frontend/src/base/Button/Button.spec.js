import { cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Primary } from './Button.stories';

afterEach(cleanup);

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Primary {...Primary.args} />, div);
});

test('matches snapshot', () => {
  const tree = renderer.create(<Primary {...Primary.args} />).toJSON();
  expect(tree).toMatchSnapshot();
});
