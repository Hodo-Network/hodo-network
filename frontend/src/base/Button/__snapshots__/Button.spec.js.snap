// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="btn btn-primary btn-md"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
  type="button"
>
  Button
</button>
`;
